### PR TITLE
add a proposal to improve Nodemetric

### PR DESCRIPTION
**Pull request by hormes**
Signed-off-by: Fansong Zeng <fanster.z@gmail.com>

### CODE REVIEW

**Comment by warmchang** [comment]
> replace NodeMetric CRD with the implementation of Aggregated APIServer.

What's the difference between this AA and [metrics-server](https://github.com/kubernetes-sigs/metrics-server)?

**Comment by hormes** [comment]
> 

we need more custom metrics.

File: docs/proposals/20220402-improve-the-efficiency-of-nodemetric.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+last-updated: 2022-04-02
+status: provisional
+---
+
+# Improve the efficiency of NodeMetrics
+
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
```
Comment (zwzhang0107) [inline docs/proposals/20220402-improve-the-efficiency-of-nodemetric.md:17]: redundant templates

File: docs/proposals/20220402-improve-the-efficiency-of-nodemetric.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+Different from webhook, the Aggregated APIServer in Koord-manager adopts the active-standby working mode, and only the leader node accepts requests:
+
+- Create independent service and healthcheck to support active/standby mode detection, only the leader node is in a healthy state.
+- NodeMetric records data in memory, which is lost when restarting, and waiting for the node to report again.
+- NodeResource supports data timeliness awareness. Nodes that have not been updated in the last period should be marked as unhealthy and start the exit process. Koord-manager restart time and data reporting recovery time, kept within the node health judgment time.
+- Koordlet finds the master node by self-service discovery, and directly reports it to Koord-manager without forwarding by apiserver, so as to avoid too high frequency to enhance the network traffic of the load balancer in front of apiserver.
```
Comment (zwzhang0107) [inline docs/proposals/20220402-improve-the-efficiency-of-nodemetric.md:69]: apiserver -> APIServer

File: docs/proposals/20220402-improve-the-efficiency-of-nodemetric.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+Different from webhook, the Aggregated APIServer in Koord-manager adopts the active-standby working mode, and only the leader node accepts requests:
+
+- Create independent service and healthcheck to support active/standby mode detection, only the leader node is in a healthy state.
+- NodeMetric records data in memory, which is lost when restarting, and waiting for the node to report again.
+- NodeResource supports data timeliness awareness. Nodes that have not been updated in the last period should be marked as unhealthy and start the exit process. Koord-manager restart time and data reporting recovery time, kept within the node health judgment time.
```
Comment (zwzhang0107) [inline docs/proposals/20220402-improve-the-efficiency-of-nodemetric.md:68]: are kept within the node health judgment time

**Pull request merged**